### PR TITLE
Web hook&description

### PR DESCRIPTION
**Pull request by LeasoftKonstantinShapoval**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,3 +21,9 @@
 
 * Added the ability for AfterPay to be available in the selected countries.
 * Replaced locally stored ginger-php library on composer library installer.
+
+** 1.9.1 ** 
+
+* Removed Webhook option from all payments.
+* Added data filter do createOrder.
```
Comment (tiutiunnykleasoft) [inline CHANGELOG.md]: 1. `to` createOrder
2. is this point is necessary to display there?

File: emspayapplepay/emspayapplepay.php [hunk 2]
```diff
@@ -152,13 +152,12 @@ public function execPayment($cart, $locale = '')
         $description = sprintf($this->l('Your order at')." %s", Configuration::get('PS_SHOP_NAME'));
         $totalInCents = EmspayHelper::getAmountInCents($cart->getOrderTotal(true));
         $currency = EmspayHelper::getPaymentCurrency();
-        $webhookUrl = Configuration::get('EMS_PAY_USE_WEBHOOK')
-            ? _PS_BASE_URL_.__PS_BASE_URI__.'modules/emspay/webhook.php'
-            : null;
+        $webhookUrl = EmspayHelper::getWebHookUrl();
+
```
Comment (tiutiunnykleasoft) [inline emspayapplepay/emspayapplepay.php]: remove empty string

File: emspaycreditcard/emspaycreditcard.php [hunk 3]
```diff
@@ -151,13 +151,12 @@ public function execPayment($cart, $locale = '')
         $description = sprintf($this->l('Your order at')." %s", Configuration::get('PS_SHOP_NAME'));
         $totalInCents = EmspayHelper::getAmountInCents($cart->getOrderTotal(true));
         $currency = EmspayHelper::getPaymentCurrency();
-        $webhookUrl = Configuration::get('EMS_PAY_USE_WEBHOOK')
-            ? _PS_BASE_URL_.__PS_BASE_URI__.'modules/emspay/webhook.php'
-            : null;
+        $webhookUrl = EmspayHelper::getWebHookUrl();
+
```
Comment (tiutiunnykleasoft) [inline emspaycreditcard/emspaycreditcard.php]: empty string

File: emspayideal/emspayideal.php [hunk 4]
```diff
@@ -161,12 +161,11 @@ public function execPayment($cart, $locale = '')
         $description = sprintf($this->l('Your order at')." %s", Configuration::get('PS_SHOP_NAME'));
         $totalInCents = EmspayHelper::getAmountInCents($cart->getOrderTotal(true));
         $currency = EmspayHelper::getPaymentCurrency();
-        $webhookUrl = Configuration::get('EMS_PAY_USE_WEBHOOK')
-            ? _PS_BASE_URL_.__PS_BASE_URI__.'modules/emspay/webhook.php'
-            : null;
+        $webhookUrl = EmspayHelper::getWebHookUrl();
+
```
Comment (tiutiunnykleasoft) [inline emspayideal/emspayideal.php]: empty string

File: emspayklarnapaylater/emspayklarnapaylater.php [hunk 5]
```diff
@@ -265,9 +265,9 @@ public function execPayment($cart, $locale)
         $description = sprintf($this->l('Your order at')." %s", Configuration::get('PS_SHOP_NAME'));
         $totalInCents = EmspayHelper::getAmountInCents($cart->getOrderTotal(true));
         $currency = EmspayHelper::getPaymentCurrency();
-        $webhookUrl = self::getWebHookUrl();
+        $webhookUrl = EmspayHelper::getWebHookUrl();
```
Comment (tiutiunnykleasoft) [inline emspayklarnapaylater/emspayklarnapaylater.php:268]: add `returnUrl` in the next line, we use in other payments variable assigned to Return URL, but in this, we take return URL directly while creating an order.

File: emspayklarnapaynow/emspayklarnapaynow.php [hunk 6]
```diff
@@ -150,13 +150,12 @@ public function execPayment($cart, $locale = '')
         $description = sprintf($this->l('Your order at')." %s", Configuration::get('PS_SHOP_NAME'));
         $totalInCents = EmspayHelper::getAmountInCents($cart->getOrderTotal(true));
         $currency = EmspayHelper::getPaymentCurrency();
-        $webhookUrl = Configuration::get('EMS_PAY_USE_WEBHOOK')
-            ? _PS_BASE_URL_.__PS_BASE_URI__.'modules/emspay/webhook.php'
-            : null;
+        $webhookUrl = EmspayHelper::getWebHookUrl();
+
```
Comment (tiutiunnykleasoft) [inline emspayklarnapaynow/emspayklarnapaynow.php]: empty string

File: emspaypayconiq/emspaypayconiq.php [hunk 7]
```diff
@@ -151,13 +151,12 @@ public function execPayment($cart, $locale = '')
         $description = sprintf($this->l('Your order at')." %s", Configuration::get('PS_SHOP_NAME'));
         $totalInCents = EmspayHelper::getAmountInCents($cart->getOrderTotal(true));
         $currency = EmspayHelper::getPaymentCurrency();
-        $webhookUrl = Configuration::get('EMS_PAY_USE_WEBHOOK')
-            ? _PS_BASE_URL_.__PS_BASE_URI__.'modules/emspay/webhook.php'
-            : null;
+        $webhookUrl = EmspayHelper::getWebHookUrl();
+
```
Comment (tiutiunnykleasoft) [inline emspaypayconiq/emspaypayconiq.php]: empty string

File: emspaypaypal/emspaypaypal.php [hunk 8]
```diff
@@ -151,13 +151,12 @@ public function execPayment($cart, $locale = '')
         $description = sprintf($this->l('Your order at')." %s", Configuration::get('PS_SHOP_NAME'));
         $totalInCents = EmspayHelper::getAmountInCents($cart->getOrderTotal(true));
         $currency = EmspayHelper::getPaymentCurrency();
-        $webhookUrl = Configuration::get('EMS_PAY_USE_WEBHOOK')
-            ? _PS_BASE_URL_.__PS_BASE_URI__.'modules/emspay/webhook.php'
-            : null;
+        $webhookUrl = EmspayHelper::getWebHookUrl();
+
```
Comment (tiutiunnykleasoft) [inline emspaypaypal/emspaypaypal.php]: empty string

File: emspaytikkiepaymentrequest/emspaytikkiepaymentrequest.php [hunk 9]
```diff
@@ -150,13 +150,12 @@ public function execPayment($cart, $locale = '')
         $description = sprintf($this->l('Your order at')." %s", Configuration::get('PS_SHOP_NAME'));
         $totalInCents = EmspayHelper::getAmountInCents($cart->getOrderTotal(true));
         $currency = EmspayHelper::getPaymentCurrency();
-        $webhookUrl = Configuration::get('EMS_PAY_USE_WEBHOOK')
-            ? _PS_BASE_URL_.__PS_BASE_URI__.'modules/emspay/webhook.php'
-            : null;
+        $webhookUrl = EmspayHelper::getWebHookUrl();
+
```
Comment (tiutiunnykleasoft) [inline emspaytikkiepaymentrequest/emspaytikkiepaymentrequest.php]: empty string

File: emspaywechat/emspaywechat.php [hunk 10]
```diff
@@ -150,13 +150,12 @@ public function execPayment($cart, $locale = '')
         $description = sprintf($this->l('Your order at')." %s", Configuration::get('PS_SHOP_NAME'));
         $totalInCents = EmspayHelper::getAmountInCents($cart->getOrderTotal(true));
         $currency = EmspayHelper::getPaymentCurrency();
-        $webhookUrl = Configuration::get('EMS_PAY_USE_WEBHOOK')
-            ? _PS_BASE_URL_.__PS_BASE_URI__.'modules/emspay/webhook.php'
-            : null;
+        $webhookUrl = EmspayHelper::getWebHookUrl();
+
```
Comment (tiutiunnykleasoft) [inline emspaywechat/emspaywechat.php]: empty string

**Review by tiutiunnykleasoft — COMMENTED** [review]
Great work 👍 , left some notices and also had a few questions :
1. Return URL for KP Later set order status as `completed` when must be `canceled`.
2. If I try to pay using Afterpay, seeing the only a white screen.

Basically working.

**Comment by LeasoftKonstantinShapoval** [comment]
Updated.

**Pull request merged**